### ISSUE TITLE
Schema registry required?

### ISSUE THREAD

**Issue by jwleath** [bitnami/bitnami-docker-kafka#213]
Image: bitnami/kafka:3.0.0
docker: 20.10.11
docker-compose: 1.17.1
Oracle VirtualBox: 6.1.14

I’m writing a producer client that sends JSON class objects but it fails because the “schema.registry.url” parameter is not set to a valid Schema Registry URL. It says this is required. Is there a way to get around this requirement for development? Note I can use kafkacat just fine with strings. The JSON class is compiled into the producer and consumer so the registry is not necessary.  I would like to be able to run without it.  If it’s required for JSON objects, what registry do you recommend?  And if the registry is required this should be mentioned in the README along with steps to install one.

Error: ConnectionRefused, caused by org.apache.kafka.common.config.ConfigException: Missing required configuration “schema.registry.url” which has no default value.

**Comment by carrodher** [comment]
It seems it is not an issue related to the bitnami Kafka container image but about how the application is being used.

For information regarding the application itself or customization of the content within the application, we highly recommend checking forums and user guides made available by the project behind the application.

That said, I will keep this ticket open until the stale bot closes it just in case someone from the community adds some valuable info.

**Comment by jwleath** [comment]
The problem was simply that I needed to run a schema-registry node. I'm surprised that a registry is required to send JSON messages but apparently that's the case. The bitnami kafka README makes no mention of this, but I was able to figure it out with the help of sample Confluent docker-compose files such as this one:
https://gist.github.com/lucasjellema/c06f8a790114396f11eadd10434d9b7e
So I was able to get it working with Confluent. Thanks anyway.
Documentation problems are also bugs. There may be no problem with your image but if someone is having trouble using it, that should also be a concern that you would help with rather than rushing to close the issue.
